### ISSUE TITLE
Sort clientList array before displaying table (may need to do when delete client and add client as well.)

### ISSUE THREAD

**Issue by xscottxbrownx** [codeforpdx/PASS#220]
### Issue:

**When `ClientListTable` renders, it shows clients in random order. Happens when adding/deleting clients as well.**

---

### Direction:

First thought on approach:
- make a clientList state and setter
- do a `.sort()` of the array (or copy of the array)
- set that sorted array to be  the clientList state
- have a `useEffect()` with this clientList state as it's dependency array

Similar to Inbox/Outbox are sorted

---

### Future PR thoughts on this:

Could have the table be sortable by user in other ways (name, priority, client "type", etc)

**Comment by xscottxbrownx** [comment]
@russfraze is pretty much done with this. Just waiting on Hugh to give them access to the GitHub here.

**Comment by leekahung** [comment]
Oh, that sounds great! Think we could have him have this things in a branch ready for a PR to Development once he gains access.

**Comment by xscottxbrownx** [comment]
> Oh, that sounds great! Think we could have him have this things in a branch ready for a PR to Development once he gains access.

Already there.   Hugh replied with a green checkmark emoji on the github access request, but I still can't seem to assign this issue to the correct Russell.

**Comment by russfraze** [comment]
I'm attempting to write a test for this. Here is what I have: 
![Screen Shot 2023-07-11 at 11 47 59 AM](https://github.com/codeforpdx/PASS/assets/81979567/7eaf4ccd-3965-4106-8401-2ff7e25ed11d)

I'm new to testing, so I could use a little guidance. 

I see the test script is listed as 'vitest' not a jest related script, but when I try to run npm vitest its an unknown command. If someone could give me a few pointers on the process I need to use, that would be great. 

I'm fine with the code being merged as is, but I'd love to know how to do this successfully if its useful.

**Comment by leekahung** [comment]
We got a npm script `test` that that already includes Vitest. It should be `npm run test <name-of-file>.test.js`

**Comment by russfraze** [comment]
![Screen Shot 2023-07-12 at 11 28 11 AM](https://github.com/codeforpdx/PASS/assets/81979567/eeec6985-4e47-435b-ace1-4cccb2f47614)
This test passes. Still looking for feedback on my approach to this.

**Comment by leekahung** [comment]
> ![Screen Shot 2023-07-12 at 11 28 11 AM](https://github.com/codeforpdx/PASS/assets/81979567/eeec6985-4e47-435b-ace1-4cccb2f47614)
> This test passes. Still looking for feedback on my approach to this. 

Personally, I think this test looks good and could probably be included to the PR for merging. Just needs to be cleaned up.

Could be formatted such that you return what is expected in a const like results and use that for expect.

So


```
const results = testArray.sort(comparePerson)

expect(results).toEqual(sortedArray)
```

Also the expect should be inside the `it`

**Comment by russfraze** [comment]
Thanks. I've made those changed and updated the branch.

**Comment by leekahung** [comment]
Oh, it would seem I'm a bit mistaken over this unit test. Should be implemented more as a snapshot (like what Scott has recently done with his PR for the delete client modal) rather than an individual unit test for a function. I saw the comments in the PR, I think Tim would be able to help you out tomorrow!

**Comment by leekahung** [comment]
PR for issue (PR #275) has been merged into Development. Closing issue.
